packages/location-field/src/index.tsx

```diff
@@ -0,0 +1,758 @@
+  suggestionCount = 3,
+  userLocationsAndRecentPlaces = [],
+  UserLocationIconComponent = UserLocationIcon
+}: LocationFieldProps): React.ReactElement => {
+  /**
+   * Gets the initial value to place in the input field.
+   */
+  const getValueFromLocation = () => {
+    const label = location?.name || "";
+    return location && !hideExistingValue ? label : "";
+  };
+
+  const listBoxId = `listbox-${optionKey}`;
+
+  const intl = useIntl();
+
+  const [activeIndex, setActiveIndex] = useState(null);
+  const [stateGeocodedFeatures, setGeocodedFeatures] = useState([]);
+  const [menuVisible, setMenuVisible] = useState(false);
+  const [stateMessage, setMessage] = useState(null);
+  const [stateValue, setValue] = useState(getValueFromLocation());
+
+  const inputRef = useRef(null);
+
+  useEffect(() => {
+    // location could be null if none is set
+    setValue(location?.name || "");
+    setGeocodedFeatures([]);
+  }, [location]);
+
+  useEffect(() => {
+    if (initialSearchResults) {
+      setGeocodedFeatures(initialSearchResults);
+      setMenuVisible(true);
+    }
+  }, [initialSearchResults]);
+
+  const geocodeAutocomplete = throttle(1000, text => {
+    if (!text) {
+      console.warn("No text entry provided for geocode autocomplete search.");
+      return;
+    }
+    getGeocoder(geocoderConfig)
+      .autocomplete({ text })
+      // TODO: Better type?
+      .then(
+        (result: {
+          features: any;
+          results: { error: { message: string } };
+        }) => {
+          let message: string;
+          // If no features found in response, default to empty array.
+          let geocodedFeatures = result && result.features;
+          if (!geocodedFeatures) {
+            // Get the Pelias error message if exists.
+            // TODO: determine how other geocoders return error messages.
+            const errorMessage =
+              result &&
+              result.results &&
+              result.results.error &&
+              result.results.error.message;
+            // If the result did not contain a list of features, add special note.
+            message = intl.formatMessage({
+              id: "otpUi.LocationField.geocoderUnreachable"
+            });
+            if (errorMessage) message += ` (${errorMessage})`;
+            geocodedFeatures = [];
+          } else if (geocodedFeatures.length === 0) {
+            message = intl.formatMessage(
+              { id: "otpUi.LocationField.noResultsFound" },
+              { input: text }
+            );
+          }
+          setGeocodedFeatures(geocodedFeatures);
+          setMessage(message);
+        }
+      )
+      .catch((err: unknown) => {
+        console.error(err);
+      });
+  });
+
+  const getFormControlClassname = () => {
+    return `${locationType}-form-control`;
+  };
+
+  const setLocation = (newLocation: Location, resultType: ResultType) => {
+    onLocationSelected(intl, { locationType, location: newLocation, resultType });
```

Comment:
Sort attributes

Reply:
Addressed in 18d5606109e7dfce7d00d42b42c8b1b6f075a0bc